core/src/services/aliyun_drive/writer.rs

```diff
@@ -108,13 +108,13 @@ impl oio::Write for AliyunDriveWriter {
         Ok(())
     }
 
-    async fn close(&mut self) -> Result<()> {
+    async fn close(&mut self) -> Result<Metadata> {
         let (Some(upload_id), Some(file_id)) = (self.upload_id.as_ref(), self.file_id.as_ref())
         else {
-            return Ok(());
+            return Ok(Metadata::default());
```

Comment:
I believe we need to set a content length for each service, at the very least. `content_length` is specifically handled in OpenDAL, where users will receive `0` if it doesn't set.